Feat: Create a github action to build the project and publish it on itch.io

oulianov opened the pull request:
Hello Maaack, this template is very cool!

I saw this issue: https://github.com/Maaack/Godot-Game-Template/issues/184 and thought I'll take a stab at it. 

Basically, this PR adds a github workflow that builds the project for Web, Linux, MacOS, and Windows. Then, it publishes it to itch.io. This workflow is triggered when creating a Github Release. I tested it on my own fork of the repo and successfully pushed files and played to the game in my browser. 

This obviously requires some setup so I added pretty extensive docs. Maybe this doc is in the "wrong" folder. Also, maybe we should add a reference to this feature somewhere else? How would you see this?

This is rather crude, but I assume this should work well for 90% of the users just looking to push something to itch or get started with CI/CD.

Feel free to tell me what can I do to get this merged.

Maaack commented on .github/workflows/docs/BuildAndPublish.md:
```diff
@@ -0,0 +1,173 @@
+   - `x.y.z` — Patch / Hotfix. Small updates, bug fixes, performance improvements, or balancing tweaks. Example: `v1.1.3` for fixing a crash or visual glitch.
+3. Publish the release. This will trigger the CICD in Github Action. Monitor its execution in the Actions tab on Github.
+
+Now, before creating a new Release, let's setup your game build and itch.io connection.
```
I found it odd to jump into this section before finishing the setup. If the intent is to sell users on the eventual ease of use before going into setup, perhaps a short version of the first line (21) can be added to the introduction.

Maaack commented on .github/workflows/docs/BuildAndPublish.md:
````diff
@@ -0,0 +1,173 @@
+PROJECT_PATH: .
+```
+
+**Edit Godot version:** By default, the workflow file is made for Godot 4.5. If you're using a different version, replace `GODOT_VERSION: 4.5` at the beginning of the file and all instances of `image: barichello/godot-ci:4.5` with your version. This workflow file uses [godot-ci](https://github.com/abarichello/godot-ci?tab=readme-ov-file) to build your game, so make sure the Godot version you're r ...
````
Missing punctuation on the last sentence.

Maaack commented on .github/workflows/docs/BuildAndPublish.md:
```diff
@@ -0,0 +1,173 @@
+EXPORT_NAME: your-game
+# NOTE: If your `project.godot` is at the repository root, set `PROJECT_PATH` to "."
+# If it's in a subdirectory, set it to the subdirectory name (e.g., "your-game")
+PROJECT_PATH: .
```
Including the comment above `EXPORT_NAME` would make more sense to me in this context.

Maaack commented on .github/workflows/docs/BuildAndPublish.md:
```diff
@@ -0,0 +1,173 @@
+
+![github secrets](./img/github-secrets.png)
+
+### 4. Setup the HTML / Playable version of your game on itch
```
This section jumps from pre-deploy setup to assuming that the user already deployed to the itch.io page. This is where I'd suggest going into creating a Release and triggering the Action. Maybe mention the ability to monitor actions and check for errors from the **Actions** tab.

Maaack commented on .github/workflows/docs/BuildAndPublish.md:
```diff
@@ -0,0 +1,173 @@
+
+![no html](./img/itch_html_missing.png)
+
+What you need to do is edit your itch project to change the **Kind of project** to be **HTML** instead of **Downloadable**.
```
This can be done in step 1 if they plan to have a web build.

Maaack reviewed (changes requested):
I tested this on a new project successfully. The documentation was quite good, though there are some changes I would request before merging it in. Thank you!

oulianov commented:
Thank you for the detailed review! Glad that you got this working. 

I changed the docs following your recommendations and also added a link to it to a relevant doc. 

Do you see other edits to be made?

Maaack commented on addons/maaacks_game_template/docs/UploadingToItchIo.md:
```diff
@@ -1,37 +1,40 @@
-The owner of the project page will also get a notification when the builds have finished processing.  
+## Automating export and publication
 
-You can re-run `./butler_manager.sh` right after an export from Godot to keep your builds synced.  
```
I just found from testing that `.github/` does not get included in release zips nor when installing from the Godot asset library. I'd recommend moving `BuildAndPublish.md` into `docs/`, the screenshots into `media/build-and-publish/` or `docs/build-and-publish/`, and copying `build-and-publish.yml` into `extras/scripts/`.

Maaack commented on addons/maaacks_game_template/docs/UploadingToItchIo.md:
```diff
@@ -1,37 +1,40 @@
 
-You can re-run `./butler_manager.sh` right after an export from Godot to keep your builds synced.  
\ No newline at end of file
+You can use Github Actions to automate these steps. Look into the `.git/workflows` folder [and this guide](../../../.github/workflows/docs/BuildAndPublish.md).
```
I just found from testing that `.github/` does not get included in release zips nor when installing from the Godot asset library.  I don't think it is critical to fix, as the content will still be available on GitHub.

If you do want to fix it, I'd recommend moving `BuildAndPublish.md` into `docs/`, the screenshots into `media/build-and-publish/` or `docs/build-and-publish/`, and copying `build-and-publish.yml` into `extras/scripts/`.

Maaack commented:
Great work! I appreciate you making the changes. I'm going to go ahead and merge this into `main`, and then make further tweaks in another PR.

I have it on my Todo to set this up for this repo, but I'd also like the push to itch.io to happen from the `demo-version` branch, which includes custom SFX and music. I hope to look into it later this week or early next.

Thanks for contributing!
